backend/src/config/config.ts

```diff
@@ -1,72 +1,82 @@
 import dotenv from "dotenv";
 import path from "path";
 
-const env = process.env.NODE_ENV || 'development';
+const env = process.env.NODE_ENV || "development";
 
-// eslint-disable-next-line 
-const config: { [index:string] : any} = new Map<string, any>();
+// eslint-disable-next-line
+const config: { [index: string]: any } = new Map<string, any>();
 
-dotenv.config({path: path.resolve(__dirname+"/../../.env")});
+dotenv.config({ path: path.resolve(__dirname + "/../../.env") });
 
 config["development"] = {
-  "databaseConfig": {
-    "username": process.env.DB_USER,
-    "password": process.env.DB_PASS,
-    "database": process.env.DB_DATABASE,
-    "host": process.env.DB_HOST,
-    "dialect": "postgres",
-    "define": {
-      "underscored": true,
-      "underscoredAll": true
-    }
-  },
-  "cookieSecret": process.env.COOKIE_SECRET,
-  "expressHost": process.env.EXPRESS_HOST || "127.0.0.1",
-  "expressPort": 5000,
-  "storageBucketName": process.env.STORAGE_BUCKET_NAME,
-  "googleSvcAcctKey": process.env.GOOGLE_SVC_ACCT_KEY,
-  "frontendUrl": process.env.FRONTEND_URL || "http://127.0.0.1:3000",
+	databaseConfig: {
+		username: process.env.DB_USER,
+		password: process.env.DB_PASS,
+		database: process.env.DB_DATABASE,
+		host: process.env.DB_HOST,
+		dialect: "postgres",
+		define: {
+			underscored: true,
+			underscoredAll: true,
+		},
+	},
+	cookieSecret: process.env.COOKIE_SECRET,
+	expressHost: process.env.EXPRESS_HOST || "127.0.0.1",
+	expressPort: 5000,
+	storageBucketName: process.env.STORAGE_BUCKET_NAME,
+	googleSvcAcctKey: process.env.GOOGLE_SVC_ACCT_KEY,
+	nextServerUrl: process.env.NEXT_SERVER_URL || "http://127.0.0.1:3000",
+	frontendUrls: [
```

Comment:
changed name and purpose. this now represents the list of CORS allowed urls, i.e. frontend urls that can access the server. This includes local urls so that we can debug or develop on the dev server in the future when it is usable.

Reply:
can we have localhost as an env too? so don't need the ||

Reply:
This variable is for the express proxy to request nextjs server for the files to send back, not for direct use from the browser. By right the frontend should be tested by turning on the server and going to the server url on the browser, and get pages through the server. You don't get hot reload with this though, so you can develop with next dev server, then test the actual behaviour with the server url.

oops. probably went off topic.

There is no need for localhost since it is only used by server to communicate. The variable is for potentially nextjs server running somewhere else and we need to point to it.